metals/src/main/scala/scala/meta/internal/metals/ReferenceProvider.scala

```diff
@@ -353,3 +321,102 @@ final class ReferenceProvider(
   }
 
 }
+
+class SymbolAlternatives(symbol: String, name: String) {
```

Comment:
Almost everything is moved except for isCompanionClass and isFieldParam

Reply:
This could just be a case class no?

Reply:
That's fine 👍🏼